File: 1_CvsPHP/src/3.bubblesort.php
/* Homework C vs. PHP Performance Analysis 3.bubblesort.php
 * 2017-2 Internet Programming
 * Written by Choe Hyeong Jin, Dept. of Computer Science and Engineering, Univ. of Seoul
 */

<!DOCTYPE html>
<meta charset="utf-8"/>

<?PHP
	function get_time(){
		list($usec, $sec) = explode(" ", microtime());
		return ((float)$usec + (float)$sec);
	}

	$LMT = 50000;
	$numbers = range($LMT, 1);
	
	echo 'Before Sort...
';
	$start = get_time();
	
	// Bubble sort...Ascending...
	echo 'BUBBLE SORTING...
';
	for($i = 0; $i < $LMT; $i++){
		for($j = 0; $j < $LMT; $j++){
			if($numbers[$i] < $numbers[$j]){
				$temp = $numbers[$i];
				$numbers[$i] = $numbers[$j];
				$numbers[$j] = $temp;
			}
		}
	}

	$end = get_time();
	echo ' After Sort...
';
	$time = $end - $start;
	echo $LMT.' Bubble Sort takes '.$time.' seconds...
';
?>

File: 1_CvsPHP/src/3.bubblesort.c
/* Homework C vs. PHP Performance Analysis 3.bubblesort.c
 * 2017-2 Internet Programming
 * Written by Lee Tae Hee, Dept. of Computer Science and Engineering, Univ. of Seoul
 */

#include <stdio.h>
#include <time.h>

int main() {
	time_t start, end;
	double gap;
	start = clock();

	int number[5000];
	int tmp;
	int lmt = 5000;

	for (int i = 0; i < lmt; i++)
		number[i] = lmt - i;

	for (int i = 0; i < lmt; i++) {
		for (int j = 0; j < lmt; j++) {
			if (number[i] < number[j]) {
				tmp = number[i];
				number[i] = number[j];
				number[j] = tmp;
			}
		}
	}

	end = clock();
	gap = (double)(end - start) / (CLOCKS_PER_SEC);
	printf("%lf ÃÊ...\n", gap);
}
